uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt5/client/paho/MqttConnectionImpl.java

```diff
@@ -112,8 +116,13 @@ public MqttSubscribeReply subscribe(long timeout, @NonNull List<Subscription> su
                 if (responseProps.getReasonString() != null) {
                     builder.setReasonString(responseProps.getReasonString());
                 }
-                if (responseProps.getUserProperties() != null) {
-                    builder.addAllProperties(convertToMqtt5Properties(responseProps.getUserProperties()));
+
```

Comment:
Have about 15 lines of common code, except maybe builder.addAllProperties() - can move to separated method to simplify